src/System.Json/tests/JsonPrimitiveTests.cs

```diff
@@ -26,6 +22,27 @@ public void ToString_Double(double value, string expected)
 
         [Theory]
         [InlineData(1.1, "1.1")]
+        [InlineData(-1.1, "-1.1")]
+        [InlineData(1e-20, "1E-20")]
+        [InlineData(1e-30, "1E-30")]
+        [SkipOnTargetFramework(~TargetFrameworkMonikers.NetFramework)]
+        public void ToString_Double_NetFramework(double value, string expected)
+        {
+            ToString(new JsonPrimitive(value), expected);
+        }
+
+        [Theory]
+        [InlineData(1.1, "1.1000000000000001")]
```

Comment:
So previously round-tripping "1.1" with "R" gave "1.1" and after this change it gives "1.1000...01"? And that is because we changed the algorithm to give better round tripping in other cases? Even if it is a strict improvement in more cases we will definitely get bug reports.. Cc @jkotas

Reply:
Does `%g` format in printf guarantee round-tripping? Can we do the same thing as `%g`?

(1.1 is printed as 1.1 by `%g` in C.)

Reply:
Yes, the IEEE spec requires you to return the full set of digits (`9` for single and `17` for double) in order to guarantee the number can always round-trip successfully.

We are going to be getting bugs either way. We already had a number of issues related to parsing, formatting, and round-tripping values behaving incorrectly: dotnet/coreclr#1316, dotnet/coreclr#3313, dotnet/coreclr#13106, dotnet/coreclr#13615, dotnet/coreclr#16783, dotnet/coreclr#17467 (with even more issues on stackoverflow, uservoice, etc)

Many of these issues have caused people to write their own implementations (such as Roslyn) or provide their own workarounds (such as various bits of serialization code in CoreFX).

For round-tripping specifically, any effort to provide "prettier" numbers will slow down the implementation and is going to be prone to bugs (as was the case already). However, as part of the updates to make things more IEEE compliant, users will have expanded capabilities that will let them get correct results in the format they desire.